src/components/modules/InfoBox/index.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { PureComponent } from 'react';
+import { StyleSheet, View, Text, ImageBackground, Dimensions } from 'react-native';
+import { ModuleBox } from '@/components/modules';
+import { Image, Container, HtmlText } from '@/components/primitives';
+import { Responsive } from '@/components/utils';
+import { COLOR_MAP } from '@/constants';
+import { BREAKPOINTS } from '@/constants/index';
+import { InfoBoxData, OFFSET_DIRECTION } from '@/types';
+
+interface Props {
+    data: InfoBoxData[];
+}
+
+type State = {
+    readText: string;
+    width: number;
+    height: number;
+    windowWidth: number;
+    imgHeight: number;
+}
+
+export default class InfoBox extends PureComponent<Props, State> {
+    state = {
+        readText: 'Read More',
+        width: 0,
+        height: 0,
+        windowWidth: 0,
+        imgHeight: 0,
+    }
+
+    private mobileInfoBox = React.createRef;
+
+    toggleReadText = (e: any) => {
+        e.stopPropagation();
+        const { readText } = this.state;
+        let toggleText = readText === 'Read Less' ? 'Read More' : 'Read Less';
+
+        this.setState({
+            readText: toggleText
+        })
+    }
+
+    // On Layout
+    // Get the dimensions of the element holding the tag description. Use that number to help determine height of the entire InfoBox module. Will need to add additional px to the number obtained by this function.
+    // Get the window width. Use that number in conjuction with the tag description dims to determine height of background image. Will need to subtract pts.
+    getHeight = (e: any) => {
+        let w = e.nativeEvent.layout.width;
+        let h = e.nativeEvent.layout.height;
+        let windowWidth = Dimensions.get('window').width;
+        let imgHeight = (windowWidth * h)/w; 
+        // Convert background height to react native's units of measurement for images
+        let finalImgHeight = (imgHeight * 100)/windowWidth; 
+
+        // On initial render, set state for h x w of description box, screen width, and background image height
+        if (this.state.imgHeight === 0) {
+            this.setState({ 
+
+                width: w,
+                height: h,
+                windowWidth: windowWidth,
+                imgHeight: finalImgHeight,
+            })
+        } else { // When toggling 'read more/read less' just update h x w for description box.
+            this.setState({ 
+                width: w,
+                height: h,
+            })
+        }
+    }
+
+    renderDesktop = () => {
+        const { data } = this.props;
```

Comment:
can you create a getInfo function that extracts all the necessary data from the prop? The call would look something like:
`
const { imageSrc, tag, description } = getInfo(data);
`

It just makes it easier to change the data structure later if we need to.

Reply:
Made a getInfo() function. Needs review.

Reply:
Removed getInfo() function.